app/src/main/java/com/android/shelfLife/model/user/UserRepositoryFirestore.kt

```diff
@@ -0,0 +1,185 @@
+    userCollection.document(currentUser.uid).update(fieldName, value).await()
+
+    // Update local _user variable
+    val currentUserData = _user.value ?: User(uid = currentUser.uid, username = "", email = "")
+    val updatedUserData =
+        when (fieldName) {
+          "username" -> currentUserData.copy(username = value as String)
+          "email" -> currentUserData.copy(email = value as String)
+          else -> currentUserData
+        }
+    _user.value = updatedUserData
+  }
+
+  // Helper function to update array fields and local user data
+  private suspend fun updateArrayField(
+      fieldName: String,
+      value: String,
+      operation: ArrayOperation
+  ) {
+    val currentUser = auth.currentUser ?: throw Exception("User not logged in")
+    val updateValue =
+        when (operation) {
+          ArrayOperation.ADD -> FieldValue.arrayUnion(value)
+          ArrayOperation.REMOVE -> FieldValue.arrayRemove(value)
+        }
+    userCollection.document(currentUser.uid).update(fieldName, updateValue).await()
+
+    // Update local _user variable
+    val currentUserData = _user.value ?: User(uid = currentUser.uid, username = "", email = "")
+    val updatedArray =
+        when (fieldName) {
+          "householdUIDs" -> {
+            val currentList = currentUserData.householdUIDs
+            when (operation) {
+              ArrayOperation.ADD -> currentList + value
+              ArrayOperation.REMOVE -> currentList - value
+            }
+          }
+          "recipeUIDs" -> {
+            val currentList = currentUserData.recipeUIDs
+            when (operation) {
+              ArrayOperation.ADD -> currentList + value
+              ArrayOperation.REMOVE -> currentList - value
+            }
+          }
+          else -> emptyList()
+        }
+    val updatedUserData =
+        when (fieldName) {
+          "householdUIDs" -> currentUserData.copy(householdUIDs = updatedArray)
+          "recipeUIDs" -> currentUserData.copy(recipeUIDs = updatedArray)
+          else -> currentUserData
+        }
+    _user.value = updatedUserData
+  }
+
+  private enum class ArrayOperation {
+    ADD,
+    REMOVE
+  }
+
+  override suspend fun addHouseholdUID(householdUID: String) {
+    updateArrayField("householdUIDs", householdUID, ArrayOperation.ADD)
+  }
+
+  override suspend fun deleteHouseholdUID(uid: String) {
+    updateArrayField("householdUIDs", uid, ArrayOperation.REMOVE)
+  }
+
+  override suspend fun addRecipeUID(recipeUID: String) {
+    updateArrayField("recipeUIDs", recipeUID, ArrayOperation.ADD)
+  }
+
+  override suspend fun deleteRecipeUID(uid: String) {
+    updateArrayField("recipeUIDs", uid, ArrayOperation.REMOVE)
+  }
+
+  override suspend fun deleteInvitationUID(uid: String) {
+    // Update Firestore
+    updateArrayField("invitationUIDs", uid, ArrayOperation.REMOVE)
+    // No need to update local _user or _invitations, as listener will handle it
+  }
+
```

Comment:
## Tests
Missing tests for RecipeUIDs functions add and delete. You can make some new tests that are rather similar to the Household ones you already made